File: sdk/node/src/api/user.learning/user.learning.model.ts

import { IsString, IsNumber, IsOptional, IsBoolean, IsUUID } from 'class-validator';

export class UserLearningCreateModel {
    @IsOptional()
    @IsUUID()
    id?: string;  
    @IsString()
    UserId: string;
    @IsString()
    ActionId: string;
    @IsString()
    ProgressStatus: string;
    @IsNumber()
    PercentageCompletion: number;

}

export class UserLearningUpdateModel {
    @IsOptional()
    @IsString()
    UserId?: string;
    @IsOptional()
    @IsString()
    ActionId?: string;
    @IsOptional()
    @IsString()
    ProgressStatus?: string;
    @IsOptional()
    @IsNumber()
    PercentageCompletion?: number;

}

export class UserLearningSearchParams {
    @IsString()
    name: string;
}

export class ApiResponse<T> {
    Data: T;

    @IsOptional()
    @IsString()
    Message?: string;

    @IsOptional()
    @IsBoolean()
    Success?: boolean;
}

File: sdk/dotnet/lmsservice_db/src/api/user.learning/UserLearningModel.cs

using System.ComponentModel.DataAnnotations;

namespace lmsservice_db.src.api.user.learning.user.learningmodel;
public class UserLearningCreateModel
{
        [Required]
    [StringLength(64, MinimumLength = 0)]
    public string UserId { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string ActionId { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string ProgressStatus { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public int PercentageCompletion { get; set; } = 0;


}

public class UserLearningUpdateModel
{
        [Required]
    [StringLength(64, MinimumLength = 0)]
    public string UserId { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string ActionId { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public string ProgressStatus { get; set; } = string.Empty;

    [Required]
    [StringLength(64, MinimumLength = 0)]
    public int PercentageCompletion { get; set; } = 0;


}

public class UserLearningSearchFilters
{
        [StringLength(64, MinimumLength = 0)]
    public string UserId { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public string ActionId { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public string ProgressStatus { get; set; } = string.Empty;

    [StringLength(64, MinimumLength = 0)]
    public int PercentageCompletion { get; set; } = 0;


}

public class ApiResponse
{
    [Required]
    public Dictionary<string, object>? Data { get; set; }

    [StringLength(500, MinimumLength = 2)]
    public string Message { get; set; } = string.Empty;

    public bool? Success { get; set; }
}

